pkg/controller/replicaset/replica_set.go

```diff
@@ -794,14 +794,19 @@ func (rsc *ReplicaSetController) syncReplicaSet(ctx context.Context, key string)
 	if manageReplicasErr != nil {
 		return manageReplicasErr
 	}
-	// Resync the ReplicaSet after MinReadySeconds as a last line of defense to guard against clock-skew.
+	// Plan the next availability check as a last line of defense against queue preemption and guard against clock-skew.
 	if updatedRS.Spec.MinReadySeconds > 0 &&
 		updatedRS.Status.ReadyReplicas == *(updatedRS.Spec.Replicas) &&
 		updatedRS.Status.AvailableReplicas != *(updatedRS.Spec.Replicas) {
-		nextSyncInSeconds = ptr.To(int(updatedRS.Spec.MinReadySeconds))
+		if nextCheck := controller.FindMinNextPodAvailabilityCheck(activePods, updatedRS.Spec.MinReadySeconds, rsc.clock.Now()); nextCheck != nil {
+			nextSyncDuration = ptr.To(*nextCheck + time.Second) // Add a second to avoid a clock-skew.
+		} else {
+			// fall back to MinReadySeconds
```

Comment:
I think this warrants a bit more of a comment. The current is self-explanatory as it is in "else".

IIUC, the fallback is just a safeguard, but should not be needed "in principle". So maybe:

```
// Safeguard fallback to MinReadySeconds. It might be used in case we are waiting for some 
// Pods to be created or scheduled while all other Pods are already available.
```

Reply:
I have changed the logic in this place. PTAL, anyway I am still keeping the MinReadySeconds as a fallback.

There is actually no explanation for including it, except as a safeguard against a mistake in our time handling code.